src/_gettsim/policy_environment.py

```diff
@@ -620,18 +619,24 @@ def transfer_dictionary(remaining_dict, new_dict, key_list):
         return remaining_dict
     else:
         # Now remaining dict is just a scalar
-        set_by_path(new_dict, key_list, remaining_dict)
+        new_dict = tree_update(tree=new_dict, tree_path=key_list, value=remaining_dict)
     return new_dict
 
 
-def get_by_path(data_dict, key_list):
-    """Access a nested object in root by item sequence."""
-    return reduce(operator.getitem, key_list, data_dict)
-
-
-def set_by_path(data_dict, key_list, value):
-    """Set a value in a nested object in root by item sequence."""
-    get_by_path(data_dict, key_list[:-1])[key_list[-1]] = value
+def _fail_if_name_of_last_branch_not_leaf_name_of_function(
```

Comment:
```suggestion
def _fail_if_name_of_last_branch_element_not_leaf_name_of_function(
```
I'd think of "branch" == "tree path". Please adjust below, too.